server/play/deal.go

```diff
@@ -0,0 +1,106 @@
+package play
+
+import (
+	"errors"
+	"log"
+
+	"github.com/joshprzybyszewski/cribbage/model"
+	"github.com/joshprzybyszewski/cribbage/server/interaction"
+)
+
+var _ PhaseHandler = (*dealingHandler)(nil)
+
+type dealingHandler struct{}
+
+func (*dealingHandler) Start(g *model.Game, pAPIs map[model.PlayerID]interaction.Player) error {
+	// Ensure all of the players hands are cleared before we start dealing
+	for pID := range g.Hands {
+		g.Hands[pID] = g.Hands[pID][:0]
+	}
+
+	// shuffle the deck at least once
+	g.Deck.Shuffle()
+
+	addPlayerToBlocker(g, g.CurrentDealer, model.DealCards, pAPIs, ``)
+
+	return nil
+}
+
+func (*dealingHandler) HandleAction(g *model.Game, action model.PlayerAction, pAPIs map[model.PlayerID]interaction.Player) error {
+	if err := validateAction(g, action, model.DealCards); err != nil {
+		return err
+	}
+
+	if action.ID != g.CurrentDealer {
+		return errors.New(`Wrong player is dealing`)
+	}
+
+	da, ok := action.Action.(model.DealAction)
+	if !ok {
+		return errors.New(`tried dealing with a different action`)
+	}
+
+	if da.NumShuffles <= 0 {
+		addPlayerToBlocker(g, g.CurrentDealer, model.DealCards, pAPIs, `Need to shuffle at least once`)
+		return nil
+	}
+
+	if len(g.BlockingPlayers) != 1 {
+		log.Printf("Expected one blocker for deal, but had: %+v\n", g.BlockingPlayers)
+	}
+	removePlayerFromBlockers(g, action)
+
+	for i := 0; i < da.NumShuffles; i++ {
+		// shuffle
+		g.Deck.Shuffle()
+	}
+
+	// deal
+	if err := deal(g, pAPIs); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func deal(g *model.Game, pAPIs map[model.PlayerID]interaction.Player) error {
+	// Get the order of players we need to deal to
+	pIDs := playersToDealTo(g)
+
+	// Define how many cards we need to deal and the hand size
+	handSize := 6
+	switch len(pIDs) {
+	case 3, 4:
+		handSize = 5
+	}
+	numCardsToDeal := handSize * len(pIDs)
+
+	// empty out the current hands if there are cards there
+	for id, h := range g.Hands {
+		if len(h) > 0 {
+			g.Hands[id] = make([]model.Card, 0, handSize)
+		}
```

Comment:
Instead, of this, we should change the `Start` method above in this file: https://github.com/joshprzybyszewski/cribbage/pull/13/files#diff-1148df7c25d212c06a90cb77ca24c95cR17-R19

The `Start` methods should be responsible for setting up any state before entering the given phase; in this case, the Deal phase should clear out every players hand. I guess we need to `make` a new slice instead of clearing out the previous.

Reply:
@joshprzybyszewski 
Looks like the `Start` method isn't getting called before the deal phase. Still investigating this.